### PR TITLE
Sda dbt core

### PR DESCRIPTION
**Pull request by david-metaplane**

### CODE REVIEW

**Comment by david-metaplane** [comment]
```
dagster_dbt.errors.DagsterDbtCliFatalRuntimeError: Fatal error in the dbt CLI (return code 2): Encountered an error: Runtime Error fatal: Invalid --project-dir flag. Not a dbt project. Missing dbt_project.yml file
  File "/usr/local/lib/python3.9/site-packages/dagster/_grpc/server.py", line 241, in __init__
    self._loaded_repositories: Optional[LoadedRepositories] = LoadedRepositories(
  File "/usr/local/lib/python3.9/site-packages/dagster/_grpc/server.py", line 104, in __init__
    loadable_targets = get_loadable_targets(
  File "/usr/local/lib/python3.9/site-packages/dagster/_grpc/utils.py", line 53, in get_loadable_targets
    else loadable_targets_from_python_package(package_name, working_directory)
  File "/usr/local/lib/python3.9/site-packages/dagster/_core/workspace/autodiscovery.py", line 49, in loadable_targets_from_python_package
    module = load_python_module(
  File "/usr/local/lib/python3.9/site-packages/dagster/_core/code_pointer.py", line 138, in load_python_module
    return importlib.import_module(module_name)
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/workspaces/metaplane-dagster/my_dagster_project/__init__.py", line 1, in <module>
    from .repository import my_dagster_project
  File "/workspaces/metaplane-dagster/my_dagster_project/repository.py", line 8, in <module>
    from .assets.dbt_core.dbt_core import dbt_assets
  File "/workspaces/metaplane-dagster/my_dagster_project/assets/dbt_core/dbt_core.py", line 9, in <module>
    dbt_assets = load_assets_from_dbt_project(project_dir=DBT_PROJECT_PATH, profiles_dir=DBT_PROFILES)
  File "/usr/local/lib/python3.9/site-packages/dagster_dbt/asset_defs.py", line 535, in load_assets_from_dbt_project
    manifest_json, cli_output = _load_manifest_for_project(
  File "/usr/local/lib/python3.9/site-packages/dagster_dbt/asset_defs.py", line 54, in _load_manifest_for_project
    cli_output = execute_cli(
  File "/usr/local/lib/python3.9/site-packages/dagster_dbt/cli/utils.py", line 139, in execute_cli
    raise DagsterDbtCliFatalRuntimeError(messages=messages)
```

File: .github/workflows/branch_deployments.yml [hunk 1]
```diff
@@ -9,6 +15,7 @@ concurrency:
 
 env:
   DAGSTER_CLOUD_URL: "http://dagster.cloud/metaplane"
```
Comment (slopp) [inline .github/workflows/branch_deployments.yml]: should be `https://metaplane.dagster.cloud`

File: my_dagster_project/assets/dbt_core/dbt_core.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from dagster import job, ScheduleDefinition, file_relative_path, define_asset_job, AssetSelection
+#from .fivetran_simple import fivetran_assets 
+#let's see if we need this as in theory the sources in the dbt project should get mapped
+from dagster_dbt import load_assets_from_dbt_project
+
+DBT_PROJECT_PATH = file_relative_path(__file__, "../../../metaplane-dbt")
```
Comment (slopp) [inline my_dagster_project/assets/dbt_core/dbt_core.py]: tweak this to be `../../dbtproj`

**Review by slopp — APPROVED** [review]
looks like it should work, but two minor changes

File: .github/workflows/branch_deployments.yml (hunk 1 above)
Comment (david-metaplane) [reply .github/workflows/branch_deployments.yml]: ok should it also be this way for the deploy.yml file? I assume so

**Comment by david-metaplane** [comment]
@slopp getting this error from the branch deployment:

```
dagster_dbt.errors.DagsterDbtCliFatalRuntimeError: Fatal error in the dbt CLI (return code 2): Encountered an error: Runtime Error fatal: Invalid --project-dir flag. Not a dbt project. Missing dbt_project.yml file
  File "/usr/local/lib/python3.8/site-packages/dagster/_grpc/server.py", line 241, in __init__
    self._loaded_repositories: Optional[LoadedRepositories] = LoadedRepositories(
  File "/usr/local/lib/python3.8/site-packages/dagster/_grpc/server.py", line 104, in __init__
    loadable_targets = get_loadable_targets(
  File "/usr/local/lib/python3.8/site-packages/dagster/_grpc/utils.py", line 53, in get_loadable_targets
    else loadable_targets_from_python_package(package_name, working_directory)
  File "/usr/local/lib/python3.8/site-packages/dagster/_core/workspace/autodiscovery.py", line 49, in loadable_targets_from_python_package
    module = load_python_module(
  File "/usr/local/lib/python3.8/site-packages/dagster/_core/code_pointer.py", line 138, in load_python_module
    return importlib.import_module(module_name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/dagster/app/my_dagster_project/__init__.py", line 1, in <module>
    from .repository import my_dagster_project
  File "/opt/dagster/app/my_dagster_project/repository.py", line 8, in <module>
    from .assets.dbt_core.dbt_core import dbt_assets, DBT_PROFILES, DBT_PROJECT_PATH, dfe_job, dfe_schedule
  File "/opt/dagster/app/my_dagster_project/assets/dbt_core/dbt_core.py", line 9, in <module>
    dbt_assets = load_assets_from_dbt_project(project_dir=DBT_PROJECT_PATH, profiles_dir=DBT_PROFILES)
  File "/usr/local/lib/python3.8/site-packages/dagster_dbt/asset_defs.py", line 535, in load_assets_from_dbt_project
    manifest_json, cli_output = _load_manifest_for_project(
  File "/usr/local/lib/python3.8/site-packages/dagster_dbt/asset_defs.py", line 54, in _load_manifest_for_project
    cli_output = execute_cli(
  File "/usr/local/lib/python3.8/site-packages/dagster_dbt/cli/utils.py", line 139, in execute_cli
    raise DagsterDbtCliFatalRuntimeError(messages=messages)
```

I take it to mean that it can't find the dbt_project.yml file in the path we set... how can we see the folder structure that dagster cloud branch deployment is using?

File: .github/workflows/branch_deployments.yml (hunk 1 above)
Comment (slopp) [reply .github/workflows/branch_deployments.yml]: yes in both places

File: my_dagster_project/assets/dbt_core/dbt_core.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from dagster import job, ScheduleDefinition, file_relative_path, define_asset_job, AssetSelection
+#from .fivetran_simple import fivetran_assets 
+#let's see if we need this as in theory the sources in the dbt project should get mapped
+from dagster_dbt import load_assets_from_dbt_project
+
+DBT_PROJECT_PATH = file_relative_path(__file__, "../../dbtproj")
```
Comment (slopp) [inline my_dagster_project/assets/dbt_core/dbt_core.py]: My bad - just pulled the image that is being built to confirm everything, and this should be `../../../dbtproj` 

(I missed the dbt_core folder earlier... )

**Pull request merged**